### ISSUE TITLE
Bug: [unbound-method] ignoreStatic does not work with Buffer.compare

### ISSUE THREAD

**Issue by vostrnad** [typescript-eslint/typescript-eslint#7893]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.ts&code=JYOwLgpgTgZghgYwgAgEIFcY2siAPSEAEwGdkBVUMADgEEoo4BPZAbwF8BYAKB6unhI0mbFADCAexAkwUdAjASobHsmQwoEgLYAKGVABcyfaADmASiMYs0VcgD0AKkd3kj5JK0AHOFAhkAAwAjTABGAORFZGDMACYAgBpIpi9gBDgAGwyWGCVIgAsUL3QoLwkSFAkYYyUwM2RfRiYyKoDrUQjQGTgQJBIAOmQAFXzgMjHcAEd0YAA3TIhwSIlkdKz6mJhQ-oRtHz8dEJhY8wD%2B1xduNTdogICAKxIL5GBvWrZhG2V2dU0tZAA5EdRACANwXZ67aRgZBHULIAC8n1E-Q02h0ANCsQAzAAWAHmcFXNTuKEyWFxRHI6Cov4YgAMWOxBKJ11JUnJjSpAG04UkjrEALqskmQjkSDIQfoZCSmHSNfokWo6do03beXwQcyE572ezIAAKUCoJCM3OQAB5VcpsfTkNj4djYvbsQA%2BJJWkQ4B3253Y7H23Gu5CC3X6nQjCZ%2BEjoDIwiYQaaZZZmgX8sKC-rmZ53ALPAACJFAQlm9P6oW2oWxBb8YBKIGQAFFgGBCsoAgBacJJAL0xLIPIBbvIIgQLyLIj1KQFFDR2Mwqoz1Z7XxjKSDADKEBQAXV%2BwgEVyylHYDgwAyAwu9jse81hzCRko4DoDGY6eOj6oL6almQIHQWhBNARJcLw3CjggGSasg8zKNaVheuIHKyPIihQESPBkjCcJUtatLopiOL4jq3BYRSxy4Yh%2BG6ACjI4iymHIQ0DA8ny5FChhpHipK0qyvKDCKsqeG3n42pEkAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJgIwHsOATWgFtE%2BABbc%2B6MAG1w2HImjRu0SABo58rPOyRKAcyYrEAZXwBDfJTJT8cRJuwBfRwF05Lp0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
interface Buffer extends Uint8Array {}

interface BufferConstructor {
  from(str: string): Buffer
  compare(buf1: Uint8Array, buf2: Uint8Array): number;
}

declare var Buffer: BufferConstructor;

const buf1 = Buffer.from('1234');
const buf2 = Buffer.from('0123');
const arr = [buf1, buf2];

console.log(arr.sort(Buffer.compare));
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/unbound-method": [
      "error",
      {
        "ignoreStatic": true
      }
    ]
  }
}
```

### Expected Result

There should be no error since I have `ignoreStatic` enabled.

### Actual Result

unbound-method error

### Additional Info

I've replicated roughly how the `Buffer` type is implemented in `@types/node` so the issue is reproducible in the playground. The code after type declarations comes directly from the `Buffer.compare` JSDoc.

The problem is likely due to the fact that `Buffer` is not as a class but a (global) variable of type `BufferConstructor`, so typescript-eslint does not treat the constructor methods as static, even though they don't belong to an instance of `Buffer`. I'm not sure if this can be solved in a way other than adding an explicit exception for `Buffer`.

**Comment by Josh-Cena** [comment]
Maybe we can say "methods on constructible interfaces"?

**Comment by bradzacher** [comment]
Yeah the funky thing is how these types are defined - technically `Buffer` is a class and technically `Buffer.compare` is a static method, however based on the types
- `Buffer` isn't actually a class - it's a `var` that has an object type with a constructor signature and
- `Buffer.compare` isn't defined as a static method - it's just a method signature on the object type.

The `ignoreStatic` option is specifically looking for members with the `static` keyword.

There are many cases of this across the built-in and nodejs types - however I'm not sure how easy it is to statically detect and classify these cases.

Note that we won't be adding a special case specifically for `Buffer` - the only acceptable solution is a general purpose one.
